src/psyclone/domain/lfric/kernel/lfric_kernel_metadata.py

```diff
@@ -0,0 +1,506 @@
+            derived type.
+        :raises ValueError: if kernel metadata is not a Fortran \
+            derived type.
+        :raises ParseError: if meta args is not a list.
+        :raises ParseError: if the metadata has an unexpected format.
+
+        '''
+        kernel_metadata = LFRicKernelMetadata()
+
+        # Ensure the Fortran2003 parser is initialised.
+        _ = ParserFactory().create(std="f2003")
+        reader = FortranStringReader(fortran_string)
+        try:
+            spec_part = Fortran2003.Derived_Type_Def(reader)
+        except Fortran2003.NoMatchError:
+            # pylint: disable=raise-missing-from
+            raise ValueError(
+                f"Expected kernel metadata to be a Fortran derived type, but "
+                f"found '{fortran_string}'.")
+
+        kernel_metadata.name = spec_part.children[0].children[1].tostr()
+
+        # the value of operates on (CELL_COLUMN, ...)
+        value = LFRicKernelMetadata._get_property(
+            spec_part, "operates_on").string
+        kernel_metadata.operates_on = value
+
+        # the value of index offset (NE, ...)
+        try:
+            value = LFRicKernelMetadata._get_property(
+                spec_part, "gh_shape").string
+            kernel_metadata.gh_shape = value
+        except ParseError:
+            kernel_metadata.gh_shape = None
+
+        # the name of the procedure that this metadata refers to.
+        kernel_metadata.procedure_name = LFRicKernelMetadata._get_property(
+            spec_part, "code").string
+
+        # meta_args contains arguments which have
+        # properties. Therefore create appropriate (ScalarArg,
+        # FieldArg, ...) instances to capture this information.
+        kernel_metadata._meta_args = LFRicKernelMetadata._get_property(
+            spec_part, "meta_args")
+        args = walk(kernel_metadata.meta_args, Fortran2003.Ac_Value_List)
+        if not args:
+            raise ParseError(
+                f"meta_args should be a list, but found "
+                f"'{str(kernel_metadata.meta_args)}' in '{spec_part}'.")
+
+        kernel_metadata._meta_args = []
+        for meta_arg in args[0].children:
+            form = meta_arg.children[1].children[0].tostr()
+            form = form.lower()
+            if form == "gh_scalar":
+                arg = ScalarArg.create_from_psyir(meta_arg)
+            elif form == "gh_operator":
+                arg = OperatorArg.create_from_psyir(meta_arg)
+            elif form == "gh_columnwise_operator":
+                arg = ColumnwiseOperatorArg.create_from_psyir(meta_arg)
+            elif "gh_field" in form:
+                vector_arg = "gh_field" in form and "*" in form
+                nargs = len(meta_arg.children[1].children)
+                intergrid_arg = False
+                if nargs == 5:
+                    fifth_arg = meta_arg.children[1].children[4]
+                    if fifth_arg.children[0].string == "mesh_arg":
```

Comment:
`intergrid_arg = fifth_arg.children[0].string == "mesh_arg"`

Reply:
you and your lhs = rhs equality skills.

Reply:
already done.